### ISSUE TITLE
Issues with LibreWolf

### ISSUE THREAD

**Issue by qualityrice** [vxiiduu/VxKex#234]
The browser LibreWolf works decently with VxKex, allowing it to run seemingly without issues at first. However it does crash after just a while, and whenever you close the browser, "LibreWolf has stopped working" shows up together with a message box saying "Librewolf could not be started. Exit code: -1073741819"

![0](https://github.com/vxiiduu/VxKex/assets/65903443/9be98e84-3b8c-4a1c-ae06-d09293778040)

**Comment by i486** [comment]
Please include VxKex log files. They are important for the author to diagnose problems and improve the application.

Steps to upload VxKex log files:

    Locate VxKex log files:
        Press the Windows key + R to open the Run dialog.
        Type in %LOCALAPPDATA%\VxKex\Logs and press Enter. This should open the directory containing the VxKex log files (if they exist).

    Clear VxKex logs (Optional):
        In the VxKex Logs folder, select all files (Ctrl+A).
        Right-click and choose "Delete" or "Move to trash" (depending on your operating system).
        If you encounter issues deleting some log files, try closing any applications that are using VxKex functionality.

    Enable VxKex and reproduce the crash:
        Follow the instructions for enabling VxKex on the application.
        Perform the actions that cause the application to crash.
        Exit/Terminate the app

    Collect VxKex logs:
        Navigate back to the VxKex Logs folder (as shown in step 1).
        Compress the log files into a single ZIP archive. You can usually do this by selecting all files, right-clicking, and choosing "Compress" or "Zip".

    Attach the log archive to your GitHub post/comment:
        To attach a file to an issue, drag and drop it into the comment box. Alternatively, you can click Paper Clip icon in the formatting bar above the comment box to browse, select, and add a file from your computer.

**Comment by qualityrice** [comment]
Sorry about that. Here's the log files from two browser sessions.
First, the browser crashed after a few minutes of browsing.
The second time, I closed the browser which also (always) results in a crash.
[LibreWolf crash Logs.zip](https://github.com/vxiiduu/VxKex/files/15253209/LibreWolf.crash.Logs.zip)

**Comment by adeii** [comment]
LibreWolf 125.x can not work with VxKex since FireFox 125.x could not, too. But you can use R3dfox 125.0.3 or this port https://github.com/adeii/supermium-portable/releases/download/F1235/librewolf-125.0.2-1-win7-x86-portable.7z

**Comment by i486** [comment]
@qualityrice 

Setting "browser.taskbar.lists.legacyBackend" to "true" fixes the "crashes after a while" issue.

**Comment by adeii** [comment]
> @qualityrice
> Setting "browser.taskbar.lists.legacyBackend" to "true" fixes the "crashes after a while" issue.
Yes, that is a solution for crash closed FF/LW. But permanent fix is to unpack broser/omni.ja (it's a ZIP), edit settings/defaults/firefox.js, pack again and rename omni.zip to omni.ja.

**Comment by qualityrice** [comment]
Seems like there are some fixes for this, or at least temporary or "half fixes", but in order to improve VxKex this issue should probably stay open right? Of course I'm not expecting a new fix/patch every time LibreWolf receives an update, but fixing this crash could help VxKex become more stable generally and maybe fix other programs too right? Or does VxKex have to "fix" each program individually for them to work? In that case it's probably a waste of time to solve every single browser issue, but I'm not sure how VxKex works. Can someone let me know? Is there any point to fixing crashes like these?

**Comment by CommonLoon102** [comment]
> Seems like there are some fixes for this, or at least temporary or "half fixes", but in order to improve VxKex this issue should probably stay open right? Of course I'm not expecting a new fix/patch every time LibreWolf receives an update, but fixing this crash could help VxKex become more stable generally and maybe fix other programs too right? Or does VxKex have to "fix" each program individually for them to work? In that case it's probably a waste of time to solve every single browser issue, but I'm not sure how VxKex works. Can someone let me know? Is there any point to fixing crashes like these?

Both. There can be app specific fixes. But in general, the fixes are covering all apps.

**Comment by adeii** [comment]
> but I'm not sure how VxKex works. Can someone let me know?

As Kernel eXtender. Every KX-file has additional functions (not included in Win 7) that load original Win 7 file and act like alternative (bridge) to that original file.  
Ex. KXBase.dll has "DisableVirtualMemory" (from Win10/11) and load Win7's kernel32.dll. Now file has the similar functions as Win10's kernel32.dll which is enough to run some Chromium-based browser.

Win8.1 do not have "DisableVirtualMemory" in its kernel32.dll and can not run browser. And there is still no Kernel EXtender for Win8.1 yet. You have 2 options - a) rename option to "VirtualAlloc" or b) use bridged file with additional functions, as like as Blaukovitch's kernel64.dll.
